fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaActivity.java

```diff
@@ -0,0 +1,210 @@
+      mViewModel.setLaoName((String) getIntent().getExtras().get(LAO_NAME));
+    }
+    setupSocialMediaHomeFragment();
+    setupNavigationBar();
+
+    // Subscribe to "lao name" string
+    mViewModel
+        .getLaoName()
+        .observe(
+            this,
+            newLaoName -> {
+              if (newLaoName != null) {
+                Objects.requireNonNull(getSupportActionBar())
+                    .setTitle(String.format("popstellar - %s", newLaoName));
+              }
+            });
+
+    // Subscribe to "open home" event
+    mViewModel
+        .getOpenHomeEvent()
+        .observe(
+            this,
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                setupSocialMediaHomeFragment();
+              }
+            });
+
+    // Subscribe to "open search" event
+    mViewModel
+        .getOpenSearchEvent()
+        .observe(
+            this,
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                setupSocialMediaSearchFragment();
+              }
+            });
+
+    // Subscribe to "open following" event
+    mViewModel
+        .getOpenFollowingEvent()
+        .observe(
+            this,
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                setupSocialMediaFollowingFragment();
+              }
+            });
+
+    // Subscribe to "open profile" event
+    mViewModel
+        .getOpenProfileEvent()
+        .observe(
+            this,
+            booleanEvent -> {
+              Boolean event = booleanEvent.getContentIfNotHandled();
+              if (event != null) {
+                setupSocialMediaProfileFragment();
+              }
+            });
+  }
+
+  public static SocialMediaViewModel obtainViewModel(FragmentActivity activity) {
+    return new ViewModelProvider(activity).get(SocialMediaViewModel.class);
+  }
+
+  @Override
+  public boolean onCreateOptionsMenu(Menu menu) {
+    getMenuInflater().inflate(R.menu.social_media_top_menu, menu);
+
+    // Get the submenu and clear its unique item. The item was needed to create the submenu
+    SubMenu laosList = menu.findItem(R.id.laos_list).getSubMenu();
+
+    // Adding all currently opened lao name to the submenu
+    mViewModel
+        .getLAOs()
+        .observe(
+            this,
+            list -> {
+              if (list != null) {
+                laosList.clear();
+                for (int i = 0; i < list.size(); ++i) {
+                  // Creating a unique id using laos_list and laos ids such that it doesn't override
+                  // them in onOptionsItemSelected
+                  laosList.add(Menu.NONE, R.id.laos_list + R.id.laos + i, i, list.get(i).getName());
```

Comment:
The third parameter seems to need a particular value from theses Menu.{CATEGORY_SYSTEM, CATEGORY_SECONDARY, CATEGORY_ALTERNATIVE, CATEGORY_CONTAINER} https://developer.android.com/reference/android/view/MenuItem#getOrder()
For the item id, didn't it work with only i ?
for example this (with i = item.getItemId() in onOptionsItemSelected) : 
```suggestion
                  laosList.add(Menu.NONE, i, Menu.CATEGORY_CONTAINER, list.get(i).getName());
```

Reply:
Yes it does work, i initially code it that way to make sure it doesn't have the same ID as laos_list but it should be fine